### PR TITLE
feat(modelreg): Keras Callback: always save model checkpoints as artifacts

### PR DESCRIPTION
**Pull request by vwrj**
https://wandb.atlassian.net/browse/WB-7345

Description
-----------
This PR toggles a boolean in order to ensure that models checkpoints are always saved as artifacts of type "model". 

So that the model checkpoints show up in the Versions tab like so:

Screen Shot 2022-07-07 at 8 14 18 PM

Testing
-------
How was this PR tested? n/a

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

**Comment by ayulockin** [comment]
If the model can be serialized in the `.h5` format and if `save_model_as_artifact` is True, the same model will be logged to W&B files and as Artifacts. 

We should either just save the model as artifacts and remove saving it as W&B files. 

or we can do 

```
if save_model_as_artifact=True :
   self. save_model = False
...

if self.save_model_model_as_artifact:
    self._save_model_as_artifact(epoch)
```

**Comment by ayulockin** [comment]
Also this if this [PR](https://github.com/wandb/client/pull/3887) gets merged, it will add better checkpointing with `WandbCallback`. Just wanted to bring this to attention as it adds some extra arguments.

**Comment by vwrj** [comment]
> If the model can be serialized in the `.h5` format and if `save_model_as_artifact` is True, the same model will be logged to W&B files and as Artifacts.
> 
> I think we should either just save the model as artifacts and remove saving it as W&B files.
> 
> or we can do
> 
> ```
> if save_model_as_artifact=True :
>    self. save_model = False
> ...
> 
> if self.save_model_model_as_artifact:
>     self._save_model_as_artifact(epoch)
> ```

Any users who have used the callback so far to save models would rely on the `.h5` files being present with wandb. If we removed this, I'm concerned that we'll break users' existing workflows. Hence a more gradual transition where we continue saving the `.h5` file while also saving the model artifact as the recommended best practice.

File: wandb/integration/keras/keras.py [hunk 1]
```diff
@@ -608,7 +608,7 @@ def on_epoch_end(self, epoch, logs={}):
             if self.save_model:
                 self._save_model(epoch)
 
-            if self.save_model_as_artifact:
+            if self.save_model and self.save_model_as_artifact:
```
Comment (tssweeney) [inline wandb/integration/keras/keras.py:611]: Small nit: might be nice to put this under the same block

**Review by tssweeney — APPROVED** [review]
Note: this is going to change the behavior of a lot of runs. Currently models will show up ion the sidebar, and after this change, that page will not be visible. We may want to make the same visualization available as part of model artifacts.

**Pull request merged**